### PR TITLE
Assessments controller - use load_resource

### PR DESCRIPTION
**Pull request by caleybrock**
Use CanCan's `load_resource` and `load_and_authorize_resource` in the assessments controller to get params instead of using custom functions.

Suggestion from https://github.com/code-dot-org/code-dot-org/pull/22816#discussion_r192304707

### CODE REVIEW

File: dashboard/app/controllers/api/v1/assessments_controller.rb [hunk 1]
```diff
@@ -1,5 +1,7 @@
 class Api::V1::AssessmentsController < Api::V1::JsonApiController
   include LevelsHelper
+  load_and_authorize_resource :section
+  load_resource :script
```
Comment (maddiedierker) [inline dashboard/app/controllers/api/v1/assessments_controller.rb]: why not `load_and_authorize_resource` here as well?

Comment (caleybrock) [reply dashboard/app/controllers/api/v1/assessments_controller.rb]: Everyone has access to a script so I didn't think it needed to be authorized? I can add it if it's best practice to always authorize.

Comment (maddiedierker) [reply dashboard/app/controllers/api/v1/assessments_controller.rb]: i *think* it's best practice, but i'm relatively new to this part of the codebase (and to using CanCan). there's already an ability set up for scripts that you should be able to use: 

https://github.com/code-dot-org/code-dot-org/blob/de0038e674ac831a9f4d799a2b42ba8b93135ac8/dashboard/app/models/ability.rb#L193-L203

**Comment by caleybrock** [comment]
Thanks for the feedback, updated!

**Pull request merged**